src/main/java/org/mdbenefits/app/preparers/HouseholdPregnancyAndDisabilityPreparer.java

```diff
@@ -0,0 +1,52 @@
+package org.mdbenefits.app.preparers;
+
+import formflow.library.data.Submission;
+import formflow.library.pdf.PdfMap;
+import formflow.library.pdf.SingleField;
+import formflow.library.pdf.SubmissionField;
+import formflow.library.pdf.SubmissionFieldPreparer;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.springframework.stereotype.Component;
+
+@Component
+public class HouseholdPregnancyAndDisabilityPreparer implements SubmissionFieldPreparer {
+
+    @Override
+    public Map<String, SubmissionField> prepareSubmissionFields(Submission submission, PdfMap pdfMap) {
+        Map<String, SubmissionField> results = new HashMap<>();
+
+        var household = (List<Map<String, Object>>) submission.getInputData().get("household");
+
+        if (household != null) {
+            AtomicBoolean householdMemberIsPregnant = new AtomicBoolean(false);
+            AtomicBoolean householdMemberHasDisability = new AtomicBoolean(false);
+            household.forEach(householdMember -> {
```

Comment:
I had considered this but wanted to avoid needing to loop over the same data twice, but to be fair, the small optimization trade off may not be worth how much more concise your recommendation would be 😅